openff/qcsubmit/tests/test_datasets.py

```diff
@@ -2610,6 +2611,25 @@ def test_optimizationdataset_qc_spec():
     assert dataset.driver == "deferred"
 
 
+def test_optimizationdataset_protocols():
+    """
+    Test adding a non-default OptimizationProtocols to an OptimizationDataset.
+    """
+
+    dataset = OptimizationDataset(
+        dataset_name="Test dataset",
+        dataset_tagline="XXXXXXXX",
+        description="XXXXXXXX",
+    )
+
+    assert dataset.protocols.trajectory == "all"
+
+    # change the protocol to only save gradient evaluations for final state
+    dataset.protocols = OptimizationProtocols(trajectory="final")
+
+    assert dataset.protocols.trajectory == "final"
+
```

Comment:
Would it be worth also adding a test to `test_submissions` that actually carries out the calculation and checks that only the final gradients are saved, or would we expect that to be covered by the QC* stack so long as our inputs have the keyword?

Reply:
I think it makes sense to do this. I'll modify the existing `test_optimization_submissions` to use `'final'` for the `trajectory` protocol and see if this gives the expected result.

Reply:
Added!